examples/inspection/plot_permutation_importance_multicollinear.py

```diff
@@ -101,28 +128,40 @@
 ax2.set_yticks(dendro_idx)
 ax2.set_xticklabels(dendro["ivl"], rotation="vertical")
 ax2.set_yticklabels(dendro["ivl"])
-fig.tight_layout()
-plt.show()
+_ = fig.tight_layout()
 
 # %%
-# Next, we manually pick a threshold by visual inspection of the dendrogram
-# to group our features into clusters and choose a feature from each cluster to
+# Next, we manually pick a threshold by visual inspection of the dendrogram to
+# group our features into clusters and choose a feature from each cluster to
 # keep, select those features from our dataset, and train a new random forest.
-# The test accuracy of the new random forest did not change much compared to
-# the random forest trained on the complete dataset.
+# The test accuracy of the new random forest did not change much compared to the
+# random forest trained on the complete dataset.
+from collections import defaultdict
+
 cluster_ids = hierarchy.fcluster(dist_linkage, 1, criterion="distance")
 cluster_id_to_feature_ids = defaultdict(list)
 for idx, cluster_id in enumerate(cluster_ids):
     cluster_id_to_feature_ids[cluster_id].append(idx)
 selected_features = [v[0] for v in cluster_id_to_feature_ids.values()]
+selected_features_names = X.columns[selected_features]
 
-X_train_sel = X_train[:, selected_features]
-X_test_sel = X_test[:, selected_features]
+X_train_sel = X_train[selected_features_names]
+X_test_sel = X_test[selected_features_names]
 
 clf_sel = RandomForestClassifier(n_estimators=100, random_state=42)
 clf_sel.fit(X_train_sel, y_train)
 print(
-    "Accuracy on test data with features removed: {:.2f}".format(
-        clf_sel.score(X_test_sel, y_test)
-    )
+    "Baseline accuracy on test data with features removed:"
+    f" {clf_sel.score(X_test_sel, y_test):.2}"
 )
+
+# %%
+# We can finally explore the permutation importance of the selected subset of
+# features:
+
+fig, ax = plt.subplots(figsize=(6, 6))
+plot_permutation_importance(clf_sel, X_test_sel, y_test, ax)
+ax.set_title("Permutation Importances on multicollinear features\n(test set)")
+ax.set_xlabel("Decrease in accuracy score")
+ax.figure.tight_layout()
+plt.show()
```

Comment:
I think it would be worth to have a new section to show how to make operate the code above with pure scikit-learn code that one would like to put in production. We don't have the tools to make dendogram for the moment so we should keep the code above.

Here, would be the code to do the processing:

```python
# %%
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import KBinsDiscretizer
from sklearn.cluster import FeatureAgglomeration

max_features = 5
pooling_func = lambda x, axis: x[:, 0]
cluster = make_pipeline(
    KBinsDiscretizer(n_bins=10, encode="ordinal", strategy="quantile"),
    FeatureAgglomeration(
        n_clusters=max_features,
        pooling_func=pooling_func,
        metric="euclidean",
        linkage="ward",
    ),
)
cluster.fit(X_train, y_train)

# %%


def importance_getter(estimator):
    """Function to use the `labels_` of `estimator` to set a weight
    of 1 for features to be kept and 0 otherwise.
    """
    # arbitrary keep the first features of a given label
    indices_columns = np.array([
        np.flatnonzero(estimator[-1].labels_ == c)[0]
        for c in range(estimator[-1].n_clusters_)
    ])
    importances = np.zeros_like(estimator[-1].labels_, dtype=float)
    importances[indices_columns] = 1
    return importances


# %%
from sklearn.feature_selection import SelectFromModel

selector = SelectFromModel(
    estimator=cluster,
    max_features=max_features,
    threshold=-np.inf,
    importance_getter=importance_getter,
)

# %%
model = make_pipeline(
    selector, RandomForestClassifier(n_estimators=100, random_state=42)
).fit(X_train, y_train)

# %%
model.score(X_test, y_test)

# %%
_, ax = plt.subplots(figsize=(6, 4))
plot_permutation_importance(model, X_train, y_train, ax=ax)
```

Reply:
We can even show that we can use `model` inside a cross-validation.

Reply:
If you want, we can delay this addition in another PR to address the original issue.